types/wlr_linux_dmabuf_v1.c

```diff
@@ -385,40 +385,35 @@ struct wlr_linux_dmabuf_v1 *wlr_linux_dmabuf_v1_from_resource(
 	return dmabuf;
 }
 
+static void linux_dmabuf_send_modifiers(struct wl_resource *resource,
+		const struct wlr_drm_format *fmt) {
+	if (wl_resource_get_version(resource) < ZWP_LINUX_DMABUF_V1_MODIFIER_SINCE_VERSION) {
+		zwp_linux_dmabuf_v1_send_format(resource, fmt->format);
```

Comment:
We previously called `send_format` once for every modifier in the format equal to `DRM_FORMAT_MOD_LINEAR`, or once if this format had no modifiers, whereas we now just `send_format` once for every format.

The old logic was oddly elaborate and [appears copied from Weston](https://gitlab.freedesktop.org/wayland/weston/-/blob/master/libweston/linux-dmabuf.c#L524), so a confirmation that this behavior change is intended would be good.

Reply:
`DRM_FORMAT_MOD_LINEAR` is a modifier, not a format.

The previous logic was pretty confusing.

- If modifiers were supported by neither the client nor the compositor: `send_format` once 
- If modifiers were not supported by the client but supported by the compositor: the list hopefully had `LINEAR` once, so we were calling `send_format` once
- If modifiers were not supported by the compositor but supported by the client: `send_modifier(INVALID)` once

Reply:
>> for every modifier in the format equal to `DRM_FORMAT_MOD_LINEAR`
>
> `DRM_FORMAT_MOD_LINEAR` is a modifier, not a format.

That's also what I tried to write. :P

> the list hopefully had LINEAR once, so we were calling send_format once

I imagined that this was an implied assumption, making for a rather odd way to do something once for every format. Appreciate the confirmation.

Reply:
I suspect the LINEAR thing is a confusion between implicit modifiers and linear layout by the way...